src/operators/Remap/avtRemapFilter.C

```diff
@@ -131,447 +138,528 @@ avtRemapFilter::SetAtts(const AttributeGroup *a)
+//      that will be used for clipping.
+//
+//  Arguments:
+//      rCell      The id of the current cell.
+//      side       The "side" of the cell as an integer.
+//
+//  Programmer: rusu1
+//  Creation:   Wed Apr  3 13:52:34 PDT 2019
+//
+// ****************************************************************************
+
+void
+avtRemapFilter::MakeClippingFunction(int rCell, int side)
+{
+    debug4 << "avtRemapFilter::MakeClippingFunction" << std::endl;
+    double cellBounds[6] = {0., 0., 0., 0., 0., 0.};
+    double origin[3] = {0., 0., 0.};
+    double normal[3] = {0., 0., 0.};
+    rg->GetCell(rCell)->GetBounds(cellBounds);
+    origin[side/2] = cellBounds[side];
+    normal[side/2] = 1.;
+
+    vtkPlane* plane = vtkPlane::New();
+    plane->SetOrigin(origin);
+    plane->SetNormal(normal);
+    vtkImplicitBoolean* funcs = vtkImplicitBoolean::New();
+    funcs->AddFunction(plane);
+    plane->Delete();
+
+    if (side == 0 || side == 1)
     {
-        rg->GetCellData()->AddArray(vars[vdx]);
-        rg->GetCellData()->SetScalars(vars[vdx]);
+        funcsArrayX.push_back(funcs);
     }
-*/    
-    avtDataRepresentation *out_dr =
-            new avtDataRepresentation(rg, 0, "", false);
-            
-std::cout << "Created out_dr:" << std::endl;
-PrintData(out_dr);
-
-std::cout << "Types and numbers of cells in sub meshes" << std::endl;
-for (std::map<std::string,int>::const_iterator iter = DEBUG_CellTypeList.begin();
-        iter != DEBUG_CellTypeList.end(); ++iter)
-{
-    if (iter->second > 0)
+    else if (side == 2 || side == 3)
     {
-        std::cout << iter->first << ": " << iter->second << std::endl;
+        funcsArrayY.push_back(funcs);
+    }
+    else //if (side == 4 || side == 5)
+    { 
+        funcsArrayZ.push_back(funcs);
     }
 }
-            
-    // --- Clean up --- //
-    avtRemapOriginalVolume->Delete();
-    //avtRemapRectilinearVolume->Delete();
-    // Delete rg?
-    // Delete vars?
-    // 
-    
-    return out_dr;
-    
+
+
+// ****************************************************************************
+//  Method: avtRemapFilter::CleanClippingFunctions
+//
+//  Purpose:
+//      Clean out all the vtkImplicitBooleans from the funcs arrays.
+//
+//  Programmer: rusu1
+//  Creation:   Mon Apr 15 10:56:53 PDT 2019
+//
+// ****************************************************************************
+
+void
+avtRemapFilter::CleanClippingFunctions()
+{
+    debug4 << "avtRemapFilter::CleanClippingFunctions" << std::endl;
+
+    for (std::vector<vtkImplicitBoolean*>::iterator iter = funcsArrayX.begin();
+         iter != funcsArrayX.end(); ++iter) {
+    	(*iter)->Delete();
+    }
+    for (std::vector<vtkImplicitBoolean*>::iterator iter = funcsArrayY.begin();
+         iter != funcsArrayY.end(); ++iter) {
+    	(*iter)->Delete();
+    }
+    for (std::vector<vtkImplicitBoolean*>::iterator iter = funcsArrayZ.begin();
+         iter != funcsArrayZ.end(); ++iter) {
+    	(*iter)->Delete();
+    }
 }
 
+
```

Comment:
Style issue with CleanClippingFunctions, opening brace should be under 'for'.

Reply:
Thanks for catching this, hard to break this habit. Fixed it.